presto-main/src/main/java/com/facebook/presto/operator/OperatorContext.java

```diff
@@ -258,19 +260,19 @@ public LocalMemoryContext newLocalSystemMemoryContext()
     // caller shouldn't close this context as it's managed by the OperatorContext
     public LocalMemoryContext localUserMemoryContext()
     {
-        return new InternalLocalMemoryContext(operatorMemoryContext.localUserMemoryContext(), memoryFuture);
+        return new UncloseableLocalMemoryContext(new InternalLocalMemoryContext(operatorMemoryContext.localUserMemoryContext(), memoryFuture));
```

Comment:
We have too many delegations now: uncloseable context -> internal context -> local user memory context. Maybe we should add constructor arguments to `InternalLocalMemoryContext` and `InternalAggregatedMemoryContext` that tells whether they should support close or not (and remove the uncloseable contexts).